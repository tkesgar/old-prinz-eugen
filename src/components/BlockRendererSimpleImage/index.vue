<template>
  <figure class="figure w-100 text-center" :class="{ 'figure-stretched': stretched }">
    <div :class="{ 'bg-light': withBackground, 'border-secondary': withBorder }">
      <img
        class="figure-img img-fluid img-height-viewport"
        :class="{ 'img-stretched': stretched }"
        :src="url"
        :alt="alt"
      />
    </div>
    <figcaption v-if="caption" class="figure-caption">
      {{ caption }}
    </figcaption>
  </figure>
</template>

<script>
export default {
  props: {
    data: {
      type: Object,
      required: true
    }
  },
  computed: {
    url () {
      return this.data.url
    },
    caption () {
      return this.data.caption
    },
    withBorder () {
      return this.data.withBorder
    },
    withBackground () {
      return this.data.withBackground
    },
    stretched () {
      return this.data.stretched
    },
    alt () {
      return this.caption || null
    }
  }
}
</script>

<style lang="scss" scoped>
.img-height-viewport {
  max-height: 100vh;
}

.figure-stretched {
  width: 100%;
}

.img-stretched {
  width: 100%;
  height: auto;
  max-height: unset;
}
</style>
