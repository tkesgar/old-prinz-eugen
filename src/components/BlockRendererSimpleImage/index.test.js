import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import BlockRendererSimpleImage from './'

const url = 'https://azurlane.yo-star.com/static/bz7.bd94444.png'
const caption = 'HMS Illustrious'

describe('BlockRendererDelimiter', () => {
  const wrapper = mount(BlockRendererSimpleImage, {
    localVue,
    propsData: {
      data: {
        url,
        caption,
        withBorder: true,
        withBackground: true,
        stretched: true
      }
    }
  })

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should render <img> with src = url', () => {
    expect(wrapper.find('img').attributes('src')).toBe(url)
  })

  test('should render caption', () => {
    expect(wrapper.find('figcaption').text()).toBe(caption)
  })

  test('should render alt', () => {
    expect(wrapper.find('img').attributes('alt')).toBe(caption)
  })

  test('should add border class', () => {
    expect(wrapper.find('.border-secondary').exists()).toBe(true)
  })

  test('should add background class', () => {
    expect(wrapper.find('.bg-light').exists()).toBe(true)
  })

  test('should add stretched class', () => {
    expect(wrapper.find('.figure-stretched').exists()).toBe(true)
  })
})

describe('BlockRendererDelimiter with no boolean data provided', () => {
  const wrapper = mount(BlockRendererSimpleImage, {
    localVue,
    propsData: {
      data: { url, caption }
    }
  })

  test('should not add border class', () => {
    expect(wrapper.find('.border-secondary').exists()).toBe(false)
  })

  test('should not add background class', () => {
    expect(wrapper.find('.bg-light').exists()).toBe(false)
  })

  test('should not add stretched class', () => {
    expect(wrapper.find('.figure-stretched').exists()).toBe(false)
  })
})

describe('BlockRendererDelimiter with empty caption', () => {
  const wrapper = mount(BlockRendererSimpleImage, {
    localVue,
    propsData: {
      data: { url, caption: '' }
    }
  })

  test('should not render caption', () => {
    expect(wrapper.find('figcaption').exists()).toBe(false)
  })

  test('should not add alt attribute', () => {
    expect(wrapper.find('img').attributes('alt')).toBeUndefined()
  })
})
