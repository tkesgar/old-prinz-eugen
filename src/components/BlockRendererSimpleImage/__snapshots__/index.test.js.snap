// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockRendererDelimiter should match snapshot 1`] = `
<figure
  class="figure w-100 text-center figure-stretched"
>
  <div
    class="bg-light border-secondary"
  >
    <img
      alt="HMS Illustrious"
      class="figure-img img-fluid img-height-viewport img-stretched"
      src="https://azurlane.yo-star.com/static/bz7.bd94444.png"
    />
  </div>
   
  <figcaption
    class="figure-caption"
  >
    
    HMS Illustrious
  
  </figcaption>
</figure>
`;
